src/test/kotlin/org/rust/cargo/CargoTestCaseBase.kt

```diff
@@ -0,0 +1,37 @@
+package org.rust.cargo
+
+import com.intellij.openapi.projectRoots.Sdk
+import com.intellij.openapi.projectRoots.impl.SdkConfigurationUtil
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.testFramework.PlatformTestCase
+import org.rust.cargo.project.RustSdkType
+
+abstract class CargoTestCaseBase : PlatformTestCase() {
+    abstract val testDataPath: String
+
+    private val sdkPath: String? get() = RustSdkType.INSTANCE.suggestHomePath()
+
+    override fun runTest() {
+        if (sdkPath == null) {
+            System.err?.println("SKIP $name: no Rust sdk found")
+            return
+        }
+        super.runTest()
+    }
+
+    override fun getTestProjectJdk(): Sdk? =
+        if (sdkPath == null)
```

Comment:
Yeah, `let` is cleaner here, fixed. I wish Rust-style `try!` was possible in Kotlin, so as to write

```Kotlin
SdkConfigurationUtil.createAndAddSDK(try!(sdkPath), RustSdkType.INSTANCE)!!
```

The nearest Kotlin alternative looks ugly:

```Kotlin
SdkConfigurationUtil.createAndAddSDK(sdkPath ?: return null, RustSdkType.INSTANCE)!!
```